### PR TITLE
Making exit flow similar to other XPK commands.

### PR DESCRIPTION
**Pull request by RoshaniN**
## Fixes / Features
- Making exit flow similar to other XPK commands, to use in Pathways integration tests.
- Ensuring that all the functions exit using xpk_exit() flow, cleanup return types.

## Testing / Documentation
Testing details.

- [ y ] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: xpk.py [hunk 1]
```diff
@@ -4243,7 +4243,7 @@ def cluster_delete(args) -> int:
   if run_gke_cluster_delete_command_code != 0:
     xpk_exit(run_gke_cluster_delete_command_code)
   xpk_print(f'GKE commands done! Cluster {args.cluster} deleted.\n')
-  return 0
+  return xpk_exit(0)
```
Comment (Obliviour) [inline xpk.py:4246]: xpk_exit doesn't return any value.

Probably what you want is without the return statement:
`xpk_exit(0)`

---

Most of the subcommand functions like cluster_delete are actually misleading to say they return int. They actually never return. Some longer term clean up exists there.

Comment (RoshaniN) [reply xpk.py]: Aah, makes sense. Thank you

**Review by Obliviour — APPROVED** [review]
Thanks for the change!

I agree we should make the major functions ( subcommand functions like xpk cluster [create / delete / list], xpk workload [xxx, yyy, zzz]) remove the return value from them
do you want to do that in this PR as well?

File: xpk.py [hunk 2]
```diff
@@ -4228,7 +4228,7 @@ def cluster_create(args) -> int:
   xpk_exit(0)
 
 
-def cluster_delete(args) -> int:
+def cluster_delete(args):
```
Comment (Obliviour) [inline xpk.py:4231]: nit`  -> None`

Comment (RoshaniN) [reply xpk.py]: Addressed and took care of the other high level functions also.

**Comment by RoshaniN** [comment]
> Thanks for the change!
> 
> I agree we should make the major functions ( subcommand functions like xpk cluster [create / delete / list], xpk workload [xxx, yyy, zzz]) remove the return value from them do you want to do that in this PR as well?

Sure, done!

**Pull request merged**